diff --git a/Telegram/SourceFiles/config.h b/Telegram/SourceFiles/config.h
index 2fa792747..e10758e04 100644
--- a/Telegram/SourceFiles/config.h
+++ b/Telegram/SourceFiles/config.h
@@ -10,6 +10,9 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "core/version.h"
 #include "settings.h"
 
+// PATCH_BY_NEBULAIM, Add by @benqi
+#define PATCH_BY_NEBULAIM
+
 constexpr str_const AppNameOld = "Telegram Win (Unofficial)";
 constexpr str_const AppName = "Telegram Desktop";
 
@@ -132,6 +135,41 @@ struct BuiltInDc {
 	int port;
 };
 
+#ifdef PATCH_BY_NEBULAIM
+
+// PATCH_BY_NEBULAIM, Add by @benqi
+// #define NEBULAIM_DC_IP4   "47.52.204.241"
+// #define NEBULAIM_DC_IP4   "182.16.84.106"
+#define NEBULAIM_DC_IP4   "47.103.102.219"
+#define NEBULAIM_DC_PORT  12345
+
+static const BuiltInDc _builtInDcs[] = {
+    { 1, NEBULAIM_DC_IP4, NEBULAIM_DC_PORT },
+    { 2, NEBULAIM_DC_IP4, NEBULAIM_DC_PORT },
+    { 3, NEBULAIM_DC_IP4, NEBULAIM_DC_PORT },
+    { 4, NEBULAIM_DC_IP4, NEBULAIM_DC_PORT },
+    { 5, NEBULAIM_DC_IP4, NEBULAIM_DC_PORT }
+};
+static const BuiltInDc _builtInDcsIPv6[] = {
+    //  { 1, "::1", _testDcPort },
+    //  { 2, "::1", _testDcPort },
+    //  { 3, "::1", _testDcPort },
+    //  { 4, "::1", _testDcPort },
+    //  { 5, "::1", _testDcPort }
+};
+static const BuiltInDc _builtInTestDcs[] = {
+    { 1, NEBULAIM_DC_IP4, NEBULAIM_DC_PORT },
+    { 2, NEBULAIM_DC_IP4, NEBULAIM_DC_PORT },
+    { 3, NEBULAIM_DC_IP4, NEBULAIM_DC_PORT }
+};
+static const BuiltInDc _builtInTestDcsIPv6[] = {
+    //  { 1, "::1", _testDcPort },
+    //  { 2, "::1", _testDcPort },
+    //  { 3, "::1", _testDcPort }
+};
+
+#else
+
 static const BuiltInDc _builtInDcs[] = {
 	{ 1, "149.154.175.50", 443 },
 	{ 2, "149.154.167.51", 443 },
@@ -160,6 +198,8 @@ static const BuiltInDc _builtInTestDcsIPv6[] = {
 	{ 3, "2001:0b28:f23d:f003:0000:0000:0000:000e", 443 }
 };
 
+#endif
+
 inline const BuiltInDc *builtInDcs() {
 	return cTestMode() ? _builtInTestDcs : _builtInDcs;
 }
